Assets/HoloToolkit/SpatialSound/Scripts/AudioInfluencerManager.cs

```diff
@@ -0,0 +1,242 @@
+                {
+                    // Apply the influencer's effect.
+                    influencer.ApplyEffect(gameObject);
+                }
+
+                // Find and remove the audio influencers that are to be removed from the audio source.
+                List<IAudioInfluencer> influencersToRemove = new List<IAudioInfluencer>();
+                foreach (IAudioInfluencer prev in previousInfluencers)
+                {
+                    MonoBehaviour mbPrev = prev as MonoBehaviour;
+
+                    // Remove influencers that are no longer in line of sight
+                    // OR
+                    // Have been disabled
+                    if (!influencers.Contains(prev) ||
+                        ((mbPrev != null) && !mbPrev.isActiveAndEnabled))
+                    {
+                        influencersToRemove.Add(prev);
+                    }
+                }
+                RemoveInfluencers(influencersToRemove);
+
+                previousInfluencers = influencers;
+                lastUpdate = now;
+            }
+	    }
+
+        /// <summary>
+        /// Removes the effects applied by specified audio influencers.
+        /// </summary>
+        /// <param name="influencers">Collection of IAudioInfluencer objects on which to remove the effect.</param>
+        private void RemoveInfluencers(List<IAudioInfluencer> influencers)
+        {
+            foreach (IAudioInfluencer influencer in influencers)
+            {
+                influencer.RemoveEffect(gameObject);
+            }
+        }
+
+        /// <summary>
+        /// Finds the IAudioInfluencer objects that are to be applied to the audio source.
+        /// </summary>
+        /// <returns>Collection of IAudioInfluencers between the user and the game object.</returns>
+        private List<IAudioInfluencer> GetInfluencers()
+        {
+            List<IAudioInfluencer> influencers = new List<IAudioInfluencer>();
+            Transform cameraTransform = CameraCache.Main.transform;
+
+            // Influencers take effect only when between the emitter and the user.
+            // Perform a raycast from the user toward the object.
+            Vector3 direction = (gameObject.transform.position - cameraTransform.position).normalized;
+            float distance = Vector3.Distance(cameraTransform.position, gameObject.transform.position);
+
+            int count = Physics.RaycastNonAlloc(cameraTransform.position,
+                                                direction,
+                                                hits,
+                                                distance,
+                                                Physics.DefaultRaycastLayers,
+                                                QueryTriggerInteraction.Ignore);
+            
+            for (int i = 0; i < count; i++)
+            {
+                IAudioInfluencer ai = hits[i].collider.gameObject.GetComponentInParent<IAudioInfluencer>();
```

Comment:
could we rename `ai` to `influencer` or something like that?

Reply:
sure. it is a holdover from the previous version of the code